# Question 7
-----

In [1]:
from scipy.integrate import quad
import math

class normal:
    def __init__(self, u, sigma):
        self.u = u
        self.sigma = sigma
        self.root2pi = math.sqrt((2 * math.pi)) 
    
    def normal_func(self, x):
        return (1 / (self.sigma * self.root2pi)) * math.exp((-1 * pow((x - self.u), 2)) / (2 * pow(self.sigma, 2)))

    def integrate(self, lower_limit, upper_limit):
        result, error = quad(self.normal_func, lower_limit, upper_limit)
        
        return result

    def locate(self, area):
        fx = 0
        x = -6
        while fx < area:
            fx = self.integrate(-6, x)
            x = x + 0.00001
            
        return x

In [2]:
z = normal(0, 1)
loc_1 = z.locate(0.2)
loc_2 = z.locate(0.15)
print(loc_1)
print(loc_2)

-0.8416100000567827
-1.0364200000563004


$$-0.84\sigma - 5.08 + \mu$$

$$\mu = 0.84\sigma + 5.08$$

$$-1.04\sigma - 5.63 + \mu = 0$$

$$-1.04\sigma - 5.63 + 0.84\sigma + 5.08 = 0$$

$$0.2\sigma = 0.55$$

$$\sigma = 2.75$$

$$\mu = (0.84 * 2.75) + 5.08$$

$$\mu = 7.4$$